File: Stravopodis_Safaric/src/uva/ql/ast/expressions/literals/BooleanLiteral.java
package uva.ql.ast.expressions.literals;

import uva.ql.ast.CodeLines;
import uva.ql.ast.value.BooleanValue;
import uva.ql.ast.visitor.VisitorInterface;

public class BooleanLiteral extends Literal{
	
	private boolean value;
	
	public BooleanLiteral(boolean _value, CodeLines _codeLines){
		super(_codeLines);
		this.value = _value;
	}
	public BooleanLiteral(CodeLines _codeLines){
		super(_codeLines);
	}
	@Override
	public BooleanValue evaluate() {
		return new BooleanValue(this.value);
	}
	@Override
	public String toString() {
		return "BooleanLiteral(" + String.valueOf(this.value) + ")";
	}
	@Override
	public <T> T accept(VisitorInterface<T> visitor) {
		return visitor.visitBooleanLiteral(this);
	}
}

File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Expressions/Literals/BooleanLiteral.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.QL.AST.Model.Statements;
using UvA.SoftCon.Questionnaire.Utilities.AST;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model.Expressions.Literals
{
    /// <summary>
    /// Represents a static, immutable boolean value.
    /// </summary>
    public class BooleanLiteral : Literal<bool>
    {
        private const string _true = "true";
        private const string _false = "false";

        public override bool IsValid
        {
            get
            {
                return Value.Equals(_true, StringComparison.OrdinalIgnoreCase)
                    || Value.Equals(_false, StringComparison.OrdinalIgnoreCase);
            }
        }

        public BooleanLiteral(string value, TextPosition position)
            : base(value, position)
        {
        }

        public override void Accept(IQLVisitor visitor)
        {
            visitor.Visit(this);
        }

        public override T Accept<T>(IQLVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override DataType GetType(IDictionary<string, DataType> symbolTable)
        {
            return DataType.Boolean;
        }

        public override bool GetValue()
        {
            if (IsValid)
            {
                return Value.Equals(_true, StringComparison.OrdinalIgnoreCase);
            }
            else
            {
                throw new InvalidOperationException("Cannot return the value of an invalid expressed boolean literal.");
            }
        }
    }
}
